### DISCUSSION TITLE
How to keep escrow of fa in contract?

### DISCUSSION THREAD

**Discussion by ajaythxkur** [aptos-labs/aptos-developer-discussions#390] [Questions] [answered]
How can we keep escrow of fa in such a way that one user deposits it and other user can withdraw that to its own account

**Comment by 0x-j** [comment, marked as answer]
You can create an object (akin to a contract account in evm) and let it escrow FA. 

```
struct FAEscrow has key {
    // the FA you want to escrow
    fa_metadata: Object<Metadata>,
    // the fungible asset store that escrow the FA
    fa_store: Object<FunigbleStore>,
    // used to generate signer for the object so you can transfer FA out\
    extend_ref: ExtendRef
}
```

**Reply by ajaythxkur** [reply]
@0xaptosj I tried the same thing here, please check this out
https://github.com/ajaythxkur/move-by-examples/blob/main/aptos-escrow/sources/aptos_escrow.move

**Reply by 0x-j** [reply]
i looked at it briefly and it looks correct, nit suggestion: i think it's better to only have 1 fa per escrow, i.e. remove `fa_b`

**Reply by ajaythxkur** [reply]
issue was something else. the answer above is correct

**Reply by 0x-j** [reply]
can you describe the issue? or paste the error

**Reply by ajaythxkur** [reply]
So the problem was here
`let fa_b = fungible_asset::withdraw(user, metadata_b, receive);`
it was giving `NOT STORE OWNER` error.
Fix:
`let fa_b = primary_fungible_store::withdraw(user, metadata_b, receive);`
